exporters/otlp/logs/src/main/java/io/opentelemetry/exporter/otlp/logs/OtlpGrpcLogExporter.java

```diff
@@ -0,0 +1,187 @@
+
+  private final MarshalerLogsServiceGrpc.LogsServiceFutureStub logsService;
+
+  private final ManagedChannel managedChannel;
+  private final long timeoutNanos;
+  private final BoundLongCounter logsSeen;
+  private final BoundLongCounter logsExportedSuccess;
+  private final BoundLongCounter logsExportedFailure;
+
+  /**
+   * Creates a new OTLP gRPC Logs Reporter with the given name, using the given channel.
+   *
+   * @param channel the channel to use when communicating with the OpenTelemetry Collector.
+   * @param timeoutNanos max waiting time for the collector to process each log batch. When set to 0
+   *     or to a negative value, the exporter will wait indefinitely.
+   * @param compressionEnabled whether or not to enable gzip compression.
+   */
+  OtlpGrpcLogExporter(ManagedChannel channel, long timeoutNanos, boolean compressionEnabled) {
+    // TODO: telemetry schema version.
+    Meter meter = GlobalMeterProvider.get().meterBuilder("io.opentelemetry.exporters.otlp").build();
+    this.logsSeen =
+        meter.counterBuilder("logsSeenByExporter").build().bind(EXPORTER_NAME_Attributes);
+    LongCounter logsExportedCounter = meter.counterBuilder("logsExportedByExporter").build();
+    this.logsExportedSuccess = logsExportedCounter.bind(EXPORT_SUCCESS_ATTRIBUTES);
+    this.logsExportedFailure = logsExportedCounter.bind(EXPORT_FAILURE_ATTRIBUTES);
+    this.managedChannel = channel;
+    this.timeoutNanos = timeoutNanos;
+    Codec codec = compressionEnabled ? new Codec.Gzip() : Codec.Identity.NONE;
+    this.logsService =
+        MarshalerLogsServiceGrpc.newFutureStub(channel).withCompression(codec.getMessageEncoding());
+  }
+
+  /**
+   * Submits all the given logs in a single batch to the OpenTelemetry collector.
+   *
+   * @param logs the list of sampled logs to be exported.
+   * @return the result of the operation
+   */
+  @Override
+  public CompletableResultCode export(Collection<LogRecord> logs) {
+    logsSeen.add(logs.size());
+    LogsRequestMarshaler request = LogsRequestMarshaler.create(logs);
+
+    final CompletableResultCode result = new CompletableResultCode();
+
+    MarshalerLogsServiceGrpc.LogsServiceFutureStub exporter;
+    if (timeoutNanos > 0) {
+      exporter = logsService.withDeadlineAfter(timeoutNanos, TimeUnit.NANOSECONDS);
+    } else {
+      exporter = logsService;
+    }
+
+    Futures.addCallback(
+        exporter.export(request),
+        new FutureCallback<ExportLogsServiceResponse>() {
+          @Override
+          public void onSuccess(@Nullable ExportLogsServiceResponse response) {
+            logsExportedSuccess.add(logs.size());
+            result.succeed();
+          }
+
+          @Override
+          public void onFailure(Throwable t) {
+            logsExportedFailure.add(logs.size());
+            Status status = Status.fromThrowable(t);
+            switch (status.getCode()) {
+              case UNIMPLEMENTED:
+                logger.log(
```

Comment:
IIUC - This won't lead to a log-loop-export because you're using a throttling logger, however this COULD potentially place a new log back into the queue to export.

Reply:
This is a good callout. What should we be doing in here, to avoid log-looping on this path. Maybe System.out is a "good enough" fallback? I don't know of a way to automatically exclude our logger here from being wired up to the exporter.